src/utils.js

```diff
@@ -52,73 +78,88 @@ export const onSwipe = (cb) => (target) => {
         const dir = absXDiff > absYDiff ? (/* h */ xDiff < 0 ? 'right' : 'left') : (/* v */ yDiff < 0 ? 'down' : 'up')
         cb(dir)
       }
-      offTouchEnd()
+      if (typeof offTouchEnd === 'function') offTouchEnd()
     })(target)
   })(target)
 
   return offTouchStart
 }
 
-export const trackTouchesForElement = (el) => {
+export const trackTouchesForElement = (el: Element): () => number => {
   let touchIds = []
   on('touchend')(({ targetTouches }) => { touchIds = targetTouches })(el)
   return () => touchIds.length
 }
 
-export const trackOngoingMouseInteraction = (el) => {
+export const trackOngoingMouseInteraction = (el: Element): () => bool => {
   let isInteracting = false
   on('mousedown')(() => { isInteracting = true })(el)
   on('mouseup')(() => { isInteracting = false })(document.body)
   return () => isInteracting
 }
 
-export const hasOngoingInteraction = (el) => {
+export const hasOngoingInteraction = (el: Element): () => bool => {
   const getOngoingTouchCount = trackTouchesForElement(el)
   const getOngoingMouseClick = trackOngoingMouseInteraction(el)
-  return () => getOngoingTouchCount() || getOngoingMouseClick()
+  return () => !!getOngoingTouchCount() || getOngoingMouseClick()
```

Comment:
does getOngoingMouseClick need !! as well?

Reply:
No, that returns a function that returns a boolean.